listeners/measured.go

```diff
@@ -46,7 +95,12 @@ func (c *wrapMeasuredConn) OnState(s http.ConnState) {
 // Responds to the "measured" message type
 func (c *wrapMeasuredConn) ControlMessage(msgType string, data interface{}) {
 	if msgType == "measured" {
-		c.Conn.ID = data.(string)
+		ctxUpdate := data.(map[string]interface{})
+		c.ctxMx.Lock()
+		for key, value := range ctxUpdate {
+			c.ctx[key] = value
+		}
+		c.ctxMx.Unlock()
 	}
 
 	// Pass it down too, just in case other wrapper does something with
```

Comment:
I realize the `if c.WrapConnEmbeddable != nil {` line below isn't in this PR @oxtoacart, but given that it's a local struct and given that we know `c.WrapConnEmbeddable` will never be `nil`, I would be tempted to just do away with the two `nil` checks.